Content.Server/Speech/Components/AddGaggedClothingComponent.cs

```diff
@@ -0,0 +1,12 @@
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Speech.Components;
+
+/// <summary>
+///     Applies <see cref="GaggedComponent"/> to user while they wear entity as clothing.
+/// </summary>
+[RegisterComponent]
+public sealed partial class AddGaggedClothingComponent : Component
```

Comment:
You should try using an inventory relay event instead (`InventoryRelayedEvent` and `InventorySystem.Relay.cs` are good places to find examples)! It'll allow you remove all this stuff and will make the entire thing a lot nicer.

Reply:
Yeah, this is good advice 👍

Reply:
With this approach, I'm able to hook into the emote lifecycle in more places.
Does it makes sense to prevent the entire emote, or leave it as the mime-like chat-without-sound I've got now?

Reply:
Emotes can also be non-verbal (for example "points at" or "stares"), which would not be affected by the gag. So I think emotes should still be possible in general

Reply:
Right, that makes sense to me.
I think I've got two separate questions:

1. Should this component prevent both the sound and the chat message for emotes that are in specific categories? eg. using the UI radial wheel could be blocked, while `@literally explodes.` would not be.
2. With the relay approach, it's a lot easier top make this more flexible such that the component indicates that it blocks `Vocal` emotes or `Hand` emotes specifically, or even particular emote prototypes by referencing their IDs. This'd enable something like "if you're wearing gloves you can't snap". Is adding this capability now desireable? To be clear, I _would not_ modify any existing items besides the muzzle to block vocal emotes.

One note on this is that handling the special Scream Action would be just kind of under the vocal emote category -- basically if you couldn't `@scream`, it'll block the Scream Action.